### PR TITLE
Add matching-brackets

### PR DESCRIPTION
**Pull request by BNAndras**
For #48in24

### CODE REVIEW

File: exercises/practice/matching-brackets/.meta/example.cairo [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                if stack.is_empty() {
+                    result = false;
+                    break;
+                }
+
+                let mut popped: Array<u8> = Default::default();
+                let mut i = 0;
+                let mut stop = stack.len() - 1;
+                while i < stop {
+                    popped.append(*stack[i]);
```
Comment (0xNeshi) [inline exercises/practice/matching-brackets/.meta/example.cairo:22]: The example doesn't compile:
```console
error: Variable was previously moved.
 --> /tmp/tmp.dfvn7nEzZd/src/lib.cairo:26:29
                let last = *stack[stack.len() - 1];
                            ^***^
note: variable was previously used here:
  --> /tmp/tmp.dfvn7nEzZd/src/lib.cairo:22:36
                    popped.append(*stack[i]);
                                   ^***^
note: Trait has no implementation in context: core::traits::Copy::<core::array::Array::<core::integer::u8>>

error: could not compile `matching_brackets` due to previous error
Error: `scarb metadata` exited with error
```
The reason this wasn't caught by the CI/CD is due to the script that just "swallows" errors in individual tests, and just marks the job as "successful". This is my bad, will address this immediately.

**Review by 0xNeshi — CHANGES_REQUESTED** [review]

File: exercises/practice/matching-brackets/.meta/example.cairo (hunk 1 above)
Comment (BNAndras) [reply exercises/practice/matching-brackets/.meta/example.cairo:22]: The test suite passes locally on 2.7.0 so I'll work on this some more.

Comment (0xNeshi) [reply exercises/practice/matching-brackets/.meta/example.cairo:22]: [CI tests are currently running with Scarb v2.6.3](https://github.com/exercism/cairo/blob/main/.github/workflows/test.yml#L41)

**Comment by 0xNeshi** [comment]
@BNAndras exercises were [updated to Cairo v2.7.1](https://github.com/exercism/cairo/pull/139), pull the most recent changes and try running the tests again

**Comment by BNAndras** [comment]
CI passes

**Review by 0xNeshi — APPROVED** [review]
Great work!

**Pull request merged**